torch_geometric/loader/node_loader.py

```diff
@@ -80,24 +79,22 @@ def __init__(
         if 'collate_fn' in kwargs:
             del kwargs['collate_fn']
 
-        # Get node type (or `None` for homogeneous graphs):
-        node_type, input_nodes = get_input_nodes(data, input_nodes)
-
         self.data = data
-        self.node_type = node_type
+        # self.node_type = node_type
         self.node_sampler = node_sampler
-        self.input_data = InputData(input_nodes, input_time)
+        self.input_nodes = get_input_nodes(data, input_nodes)
+        # self.input_data = InputData(input_nodes, input_time)
         self.transform = transform
         self.filter_per_worker = filter_per_worker
 
-        iterator = range(input_nodes.size(0))
+        iterator = range(len(self.input_nodes))
         super().__init__(iterator, collate_fn=self.collate_fn, **kwargs)
 
     def collate_fn(self, index: NodeSamplerInput) -> Any:
         r"""Samples a subgraph from a batch of input nodes."""
-        input_data: NodeSamplerInput = self.input_data[index]
+        input_nodes: NodeSamplerInput = self.input_nodes[index]
```

Comment:
Am I correct in understanding that, in the final version, `InputData` will be responsible for producing a `Dict[str, NodeSamplerInput]` (e.g. using `from_list`), and we will then sample on this dictionary by providing support for this in the sampler implementation?